test/backend/remote.js

```diff
@@ -0,0 +1,484 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+var test = require('tap').test
+
+var fake = require('../fake')
+var P = require('../../promise')
+var clientThen = require('../client-then')
+
+function emailToHex(email) {
+  return Buffer(email).toString('hex')
+}
+
+function respOk(t, r) {
+  t.equal(r.res.statusCode, 200, 'returns a 200')
+  t.equal(r.res.headers['content-type'], 'application/json', 'json is returned')
+}
+
+function respOkEmpty(t, r) {
+  t.equal(r.res.statusCode, 200, 'returns a 200')
+  t.equal(r.res.headers['content-type'], 'application/json', 'json is returned')
+  t.deepEqual(r.obj, {}, 'Returned object is empty')
+}
+
+function testNotFound(t, err) {
+  t.equal(err.statusCode, 404, 'returns a 404')
+  t.deepEqual(err.body, { message : 'Not Found' }, 'Object contains no other fields')
+}
+
+module.exports = function(cfg) {
```

Comment:
It's probably worth a short comment in this and its sibling files, to explain how to use these tests in a new backend.  It took me a few moments to work out exactly what was going on when the mysql backend calls this function.